### DISCUSSION TITLE
Format the text output of so.Text()

### DISCUSSION THREAD

**Discussion by FirefoxMetzger** [mwaskom/seaborn#3142] [Q&A] [answered]
Is it possible to apply post-processing to the value displayed by [`so.Text()`](https://seaborn.pydata.org/generated/seaborn.objects.Text.html#seaborn.objects.Text)? I want to create a "top 5 vs. bottom 5" style plot and my naive attempt is looking a bit odd:

![31333794-e870-450e-874e-26c08077d0b7](https://user-images.githubusercontent.com/4402489/201662499-0e14b800-0e52-4508-b0b3-9f6532d18192.png)

I would like to do two things: 

1. Format the value of text to something like `f"{text:.2%}"`
2. set `halign` to `"right"` if its position is above some threshold so that it appears inside the bar instead of being clipped.

Sidenote: I am a big fan of the flexibility of the objects API 🚀 though I must admit I haven't fully understood the intuition behind it yet, so it could very well be that this feature exists and that I just missed it in the docs because I've searched the wrong places.

**Comment by mwaskom** [comment, marked as answer]
Here's what I would do right now

```python
df = sns.load_dataset("glue").groupby("Model")[["Score"]].mean()
(
    so.Plot(
        df,
        x="Score",
        y="Model",
        text=df["Score"].round(1),
        halign=df["Score"] > 70,
    )
    .add(so.Bar())
    .add(so.Text(), color=df["Score"] > 70)
    .scale(color=["b", "w"])
)
```
<img width=600 src=https://user-images.githubusercontent.com/315810/201787595-c4d85504-6ed2-4323-b8a0-3467cc5791d4.png />

I think that some aspects of this will get cleaner with #3107, e.g. instead of `halign=df["Score"] > 70` you would have `halign=lambda: "Score > 70"`. and instead of `text=df["Score"].round(1)` you'd have `text=lambda: "Score.round(1)"`.

Format is a little trickier and I wouldn't say I'm against having a format property for the text mark, but there are enough annoying open questions (what is the default? how does the mapping work?) that I'm holding off for now.

**Reply by FirefoxMetzger** [reply]
Oh, I didn't realize that you can pass an ArrayLike with matching length. That's good to know!
